### ISSUE TITLE
HBBTV support at Exoplayer

### ISSUE THREAD

**Issue by hoimingw** [google/ExoPlayer#1421]
I tried to player a HBBTV 1.5 compatible DASH package on Exoplayer and found the following behaviors.
1. If video parameter sets pass-through is not set (i.e. SPS/PSS exist in the init segment only), video cannot be rendered.
2. If the video parameters set pass-through is set (i.e. SPS/PSS exist in the video payload), video can be rendered.
I suspect that it is due to Exoplayer could not find the correct decoder configuration information from the HBBTV compatible init segment, which is shared by all representations in an adaptation set.

Please advise whether Exoplayer supports HBBTV 1.5 compatible DASH package.

Exoplayer version: r1.5.6

Manifest and init segments can be downloaded from the following FTP server.
Filename:        hbbtv_PPS_woPassthrough.zip
FTP Server:	ftp.harmonicinc.com
Login Name:	hlit2720
Password:	mW5nLb6m

**Comment by AquilesCanta** [comment]
The provided file does not include the media chunks, only initialization. Also, I tried to access it again to double check but I no longer have access.

**Comment by hoimingw** [comment]
I just tested with r1.5.7 and found that the video could be rendered by serious blocking artifacts appeared occasionally. The playback behavior was similar for both with and without parameter pass-through.

I have already sent the link to the complete bug report and test asset to dev.exoplayer@gmail.com including the issue number.

Thanks,
Hoi Ming

**Comment by AquilesCanta** [comment]
I will get into this now, would you please clarify "... could be rendered by serious blocking artifacts appeared occasionally". I suppose the "by" is a "but"? I will test it soon though, and see what you mean if I can reproduce it.

**Comment by hoimingw** [comment]
Sorry it was a typo, it should be "but".

**Comment by AquilesCanta** [comment]
Sorry for the time I took. @ojw28 has found an issue that is probably the cause of the malfunctioning. The initialization segments are the same for every representation, except for the width/height parameter in the track header box. This means that every initialization segment includes four VisualSampleDescription boxes, one per representation. I have not managed to find any information about this in the MP4 standard, but in my opinion it wouldn't make too much sense to allow this, because of the ambiguity it generates. On top of that, I can say that ExoPlayer doesn't handle this.

In conclusion, I would suggest trying to change the initialization segments so that they include one VisualSampleDescription box each, that describes the specific representation they belong to. If the error is reproducible even after this, please open a new issue with the new stream data or, at least, the new initialization segments.

Again, sorry for the time this took.
